microsetta_admin/server.py

```diff
@@ -154,39 +154,67 @@ def _search(resource=None):
             return result
 
 
-@app.route('/create_project', methods=['GET', 'POST'])
-def new_project():
-    if request.method == 'GET':
-        return render_template('create_project.html',
-                               **build_login_variables())
-    elif request.method == 'POST':
-        project_name = request.form['project_name']
-        is_microsetta = request.form.get('is_microsetta', 'No') == 'Yes'
-        bank_samples = request.form.get('bank_samples', 'No') == 'Yes'
-        plating_start_date = request.form.get('plating_start_date')
-        if plating_start_date == '':
-            plating_start_date = None
-
-        status, result = APIRequest.post('/api/admin/create/project',
-                                         json={'project_name': project_name,
-                                               'is_microsetta': is_microsetta,
-                                               'bank_samples': bank_samples,
-                                               'plating_start_date':
-                                                   plating_start_date
-                                               })
-
-        if status == 201:
-            return render_template('create_project.html', message='Created!',
-                                   **build_login_variables())
+def _translate_nones(a_dict, do_none_to_str):
+    # Note: this ISN'T a deep copy. This function is NOT set up
+    # for recursing through a multi-layer dictionary
+    result = a_dict.copy()
+    for k, v in result.items():
+        if do_none_to_str and v is None:
+            result[k] = ""
+        elif not do_none_to_str and v == '':
+            result[k] = None
+    return result
+
+
+@app.route('/manage_projects', methods=['GET', 'POST'])
+def manage_projects():
+    result = None
+    if request.method == 'POST':
+        projects_uri = '/api/admin/projects'
+
+        model = {x: request.form[x] for x in request.form}
+        project_id = model.pop('project_id')
+        model['is_microsetta'] = model.get('is_microsetta', False) == 'true'
```

Comment:
Are only allowed values for is_microsetta and bank_samples the strings 'true' and 'false'?  'True' would map to false in this case, as would the boolean value True.

Reply:
> Are only allowed values for is_microsetta and bank_samples the strings 'true' and 'false'?
Actually, it's a little stranger than that: the *only* possible value for the 'is_microsetta' and 'bank_samples' inputs in the html form (which is the source of `model`) is "true".  That is because they are both represented as checkboxes, like this:
`<input type="checkbox" name="is_microsetta" id="is_microsetta" value='true' />`

Thus, if and only if the checkbox is checked, request.form contains an entry for the checkbox's name, and that entry will always have the "value" entry as its contents.  If the checkbox is NOT checked, then request.form does not contain ANY entry for the checkbox's name.  That's where the "get" with its default value comes into play :)

Reply:
I have changed the get defaults to empty string instead of False, to hopefully make this a bit easier to follow.